analyses/molecular-subtyping-MB/02-compare-classes.Rmd

```diff
@@ -114,43 +118,102 @@ compute.accuracy <- function(expected.input, observed.input){
+    
+    # if A has a consensus molecular_subtype match and B has molecular_subtype as NA
+    if(nrow(rna.id.with.molsubtype) == 1 & nrow(rna.id.without.molsubtype) == 1){
+      # propagate consensus subtype of A if it matches to either prediction in B
+      if(grepl(rna.id.with.molsubtype$molecular_subtype, rna.id.without.molsubtype$MM2S_best_fit) | 
+         grepl(rna.id.with.molsubtype$molecular_subtype, rna.id.without.molsubtype$medulloPackage_best_fit)){
+        rna.id.without.molsubtype <- rna.id.without.molsubtype %>%
+          mutate(molecular_subtype = rna.id.with.molsubtype$molecular_subtype)
+      }
+    }
+    
+    # combine both rows
+    mat <- rbind(rna.id.with.molsubtype, rna.id.without.molsubtype)
+  } 
+  return(mat)
+}
+```
+</details>
+
 <details><summary>Create consensus output + compute accuracy </summary>
 ```{r consensus_table, echo = TRUE}
 # function for merging outputs from both classifiers
-calc.consensus <- function(mm2s, medullo.classifier){
+calc.consensus <- function(mm2s, medullo.classifier, clin){
   # create df with common cols
   mm2s <- mm2s %>%
     dplyr::rename(MM2S_best_fit = best.fit) %>%
     dplyr::select(Kids_First_Biospecimen_ID, sample_id, 
-           pathology_subtype, other_molecular_findings, MM2S_best_fit)
+                  pathology_subtype, other_molecular_findings, MM2S_best_fit)
   medullo.classifier <- medullo.classifier %>%
     dplyr::rename(medulloPackage_best_fit = best.fit) %>%
     dplyr::select(Kids_First_Biospecimen_ID, sample_id, 
-           pathology_subtype, other_molecular_findings, medulloPackage_best_fit)
+                  pathology_subtype, other_molecular_findings, medulloPackage_best_fit)
   
   # join both
   consensus.mat <- mm2s %>%
     inner_join(medullo.classifier, 
                by = c('Kids_First_Biospecimen_ID', 'sample_id',
                       'pathology_subtype', 'other_molecular_findings'))
-
+  
   # create consensus molecular subtype if both classifiers agree 
   # match consensus molecular subtype to the pathology report
   consensus.mat  <- consensus.mat %>%
     mutate(molecular_subtype = ifelse(MM2S_best_fit == medulloPackage_best_fit, MM2S_best_fit, NA)) %>%
     mutate(match = str_detect(pathology_subtype, molecular_subtype))
   
+  # merge with clinical data
+  consensus.mat.out <- clin.merged %>%
+    inner_join(consensus.mat %>% 
+                 mutate(Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID), 
+               by = c('sample_id', 'Kids_First_Biospecimen_ID_RNA'))
+  
+  # resolve discrepancy with multiple sample ids
+  # apply prop.mb.subtype function on same event (`sample_id`) with the same `tumor_descriptor`
+  consensus.mat.out <- plyr::ddply(consensus.mat.out, 
+                                   .variables = c("sample_id", "tumor_descriptor"), 
+                                   .fun = function(x) prop.mb.subtype(mat = x))
+  consensus.mat.out$sample_id <- as.factor(consensus.mat.out$sample_id)
```

Comment:
Why does `sample_id` need to be a factor?

Reply:
@jaclyn-taroni Good question - I was having some issues with viewing the `sample_id` column with the DT::datatable function in the sense that the hyphens in the `7316-` ids were dynamically getting removed and I am still not sure why. After converting it to a factor, that issue is resolved. In addition to this, earlier it would give a fuzzy match so if you searched for 7316-14 it would also show 7316-1460 as well. Now, you can search for exact matches.

Reply:
Huh, the `-` thing is weird, thanks!